.github/workflows/codeql.yml

```diff
@@ -0,0 +1,143 @@
+         vulkan-validationlayers-dev                    \
+         libz-dev                                       \
+         llvm-dev                                       \
+         libfreetype6-dev                               \
+         libimgui-dev                                   \
+         protobuf-compiler-grpc                         \
+         pkg-config                                     \
+         libvulkan-volk-dev                             \
+         libvulkan-dev                                  \
+         libopengl-dev                                  \
+         libglx-dev                                     \
+         mesa-common-dev                                \
+         qtbase5-dev                                    \
+         libgtest-dev                                   \
+         libgmock-dev                                   \
+         git                                            \
+         libprotobuf-dev
+    - name: Save CCache Timestamp
+      if: ${{ matrix.language == 'cpp' }}
+      id: ccache_timestamp
+      run: echo "timestamp=$(date +%m-%d-%Y--%H:%M:%S)" >> $GITHUB_OUTPUT
+    - name: Setup CCache Files
+      if: ${{ matrix.language == 'cpp' }}
+      uses: actions/cache@v3
+      with:
+        path: .ccache
+        key: CodeQL-ccache-${{ steps.ccache_timestamp.outputs.timestamp }}
+        restore-keys: |
+          CodeQL-ccache-
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+        languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file.
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        
+        # Details on CodeQL's query packs refer to : https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#using-queries-in-ql-packs
+        # queries: security-extended,security-and-quality
+
+        
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, Go, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    # - name: Autobuild
+    #   uses: github/codeql-action/autobuild@v2
+
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun
+
+    #   If the Autobuild fails above, remove it and uncomment the following three lines. 
+    #   modify them (or add more) to build your code if your project, please refer to the EXAMPLE below for guidance.
+
+    # - run: |
+    #   echo "Run, Build Application using script"
+    #   ./location_of_script_within_repo/buildscript.sh
```

Comment:
What about this comment? Shall it be removed?

Reply:
Removed it.